src/frontends/lean/elaborator.cpp

```diff
@@ -2766,7 +2779,9 @@ static expr translate(environment const & env, local_context const & lctx, expr
 
 expr nested_elaborate(environment & env, options const & opts, metavar_context & mctx, local_context const & lctx,
                       expr const & e, bool relaxed) {
-    elaborator elab(env, opts, mctx, lctx);
+    // TODO(sullrich): pass through info_manager?
```

Comment:
@kha I think the top-level elaborator should set the thread local variable. 
So, we make sure that nested elaborators use the same `info_manager` being used by the top-level one.